### PR TITLE
Allow building without PostgreSQL or SQLite backend

### LINKED ISSUE #3629: Allow building without PostgreSQL or SQLite backend
We should allow building FerretDB without PostgreSQL and/or SQLite backend with `ferretdb_no_postgresql` and `ferretdb_no_sqlite` build tags. Both should be absent in official builds.

They should be documented at https://github.com/FerretDB/FerretDB/blob/main/build/version/version.go and tested at https://github.com/FerretDB/FerretDB/blob/main/internal/handlers/registry/registry_test.go.

### PR DESCRIPTION
**Pull request by anunayasri**
# Description

**Requirement**:  We should allow building FerretDB without PostgreSQL and/or SQLite backend with `ferretdb_no_postgresql` and `ferretdb_no_sqlite` build tags. Both should be absent in official builds.

I have added the relevant build tags in postgres & sqlite handler and registry files. The corresponding test `TestDeps` is skipped in the `main` branch. I have added placeholders for testing the above build tag changes. We can fix these once the `TestDeps` is fixed.

## Testing Strategy

1. Add required build tags in `BUILD_TAGS` in `Taskfile.yml`.
2. Build using `task build-host`.
3. Check the compiled handlers using `bin/ferretdb -h`.

### Test Results 

1. Build without Postgresql

Build tags given
image

Postgres handler is absent.
image

2. Build without Sqlite

Build tags given
image

Sqlite handler is absent.
image

Closes #3629 

## Readiness checklist

- [x] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

**Comment by AlekSi** [comment]
@anunayasri please click the CLA button in the comment above

File: build/version/version.go [hunk 1]
```diff
@@ -28,8 +28,10 @@
 // The following Go build tags (also known as build constraints) affect all builds of FerretDB,
 // including embedded usage:
 //
-//	ferretdb_debug - enables debug build (see below; implied by builds with race detector)
-//	ferretdb_hana  - enables Hana backend (alpha)
+//	ferretdb_debug			- enables debug build (see below; implied by builds with race detector)
+//	ferretdb_hana			- enables Hana backend (alpha)
+//	ferretdb_no_postgres	- disables postgres backend
```
Comment (AlekSi) [inline build/version/version.go:33]:
```suggestion
//	ferretdb_no_postgresql	- disables PostgreSQL backend
```

File: build/version/version.go [hunk 2]
```diff
@@ -28,8 +28,10 @@
...
+//	ferretdb_no_sqlite		- disables postgres backend
```
Comment (AlekSi) [inline build/version/version.go:34]:
```suggestion
//	ferretdb_no_sqlite		- disables SQLite backend
```

**Review by AlekSi — COMMENTED** [review]
Well, build tags look good, but we need to update the test

**Comment by AlekSi** [comment]
> The corresponding test TestDeps is skipped in the main branch

Why do you think so?

**Review by AlekSi — APPROVED** [review]

**Review by henvic — APPROVED** [review]

**Pull request merged**